internal/controller/progressive/progressive.go

```diff
@@ -404,13 +404,21 @@ func AssessUpgradingPipelineType(
 		WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
 		Debugf("Upgrading child is in phase %s, conditions healthy=%t, ready replicas match desired replicas=%t", upgradingObjectStatus.Phase, healthyConditions, healthyReplicas)
 
-	if analysisStatus != nil {
-		if upgradingObjectStatus.Phase == "Running" && healthyConditions && healthyReplicas && analysisStatus.Phase == argorolloutsv1.AnalysisPhaseSuccessful {
-			return apiv1.AssessmentResultSuccess, "", nil
-		}
-	}
-
+	// conduct standard health assessment first
 	if upgradingObjectStatus.Phase == "Running" && healthyConditions && healthyReplicas {
+		// if analysisStatus is set with an AnalysisRun's name, we must also check that it is in a Completed phase to declare success
+		if analysisStatus != nil && analysisStatus.AnalysisRunName != "" {
+			numaLogger.WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
+				Debugf("AnalysisRun is in phase %s", analysisStatus.AnalysisRunName)
```

Comment:
I think your Debugf statement has the wrong variable passed in

Reply:
make sure you specify in the log both the AnalysisRun name and the phase

Reply:
Ack, sorry about the mistake. Fixing that now like so:

```
numaLogger.WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
				Debugf("AnalysisRun %s is in phase %s", analysisStatus.AnalysisRunName, analysisStatus.Phase)
```